contribs/gmf/src/services/querymanager.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @fileoverview This file defines the QueryManager service, which uses the
+ * c2cgeoportal's themes to configure ngeo's query service with each layer
+ * found.
+ */
+goog.provide('gmf.QueryManager');
+
+goog.require('gmf');
+goog.require('gmf.Themes');
+goog.require('ngeo.Query');
+
+
+
+/**
+ * @constructor
+ * @param {ngeo.Query} ngeoQuery The ngeo Query service.
+ * @param {gmf.Themes} gmfThemes The gmf Themes service.
+ * @param {string} gmfWmsUrl URL to the wms service to use by default.
+ * @ngInject
+ * @ngdoc service
+ * @ngname gmfThemes
+ */
+gmf.QueryManager = function(ngeoQuery, gmfThemes, gmfWmsUrl) {
+
+  /**
+   * @type {string}
+   * @private
+   */
+  this.gmfWmsUrl_ = gmfWmsUrl;
+
+  // the themes have to be loaded at this point
+  // FIXME, should it be the responsability of this manager to do this ?
+  gmfThemes.loadThemes();
+
+  /**
+   * @type {Array.<ngeox.QuerySource>}
+   * @private
+   */
+  this.sources_ = [];
+
+  /**
+   * @type {Object.<number|string, ngeox.QuerySource>}
+   * @private
+   */
+  this.cache_ = {};
+
+  gmfThemes.getThemesObject().then(goog.bind(function(themes) {
+    // create sources for each themes
+    for (var i = 0, len = themes.length; i < len; i++) {
+      this.createSources_(themes[i]);
+    }
+    // then add them to the query service
+    ngeoQuery.addSources(this.sources_);
+  }, this));
+
+};
+
+
+/**
+ * Create and add a source for the query service from the GMF theme node if
+ * it has no children, otherwise create the sources for each child node if
+ * it has any.
+ * @param {GmfThemesNode} node
```

Comment:
Please add a description to the param. Caps and end with a point.
For your whole code